### PR TITLE
Visual Regression Testing

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
**This PR does the following:**
* Installs tools required for automated visual regression testing
* Creates example test for the full style-guide as well as one for buttons

## To Review:
* [ ] Checkout this branch
* [ ] Install [GraphicsMagick](https://github.com/webdriverio/webdrivercss#install)
  * On mac `brew install graphicsmagick`
* [ ] Run `npm install` (Yarn doesn't work yet)
* [ ] In a dedicated terminal window, run `npm start` to spin up pattern lab
* [ ] In a dedicated terminal window, run `./node_modules/.bin/chromedriver --url-base=/wd/hub --port=4444`
* [ ] To test all components, in your main terminal type `node tests/visual/visreg.js`
* [ ] To test just the card component, move to the card directory `cd components/_patterns/02-molecules/card` and run `node card.test.js`

## Todo:
* [ ] Make it compatible with Yarn
* [ ] Document how to use/create visreg tests

## Would like to:
* [x] Rename tests file to something visual regression specific
* [ ] Extract tests into individual files so that you can:
    * [x] Isolate tests per-component. ie:
    ```
    /visregtests
      /buttons
        /buttons.js
        /buttons-hover.js
      /header
        /header.js
    ```
    Or maybe even include them with the twig, scss files!!!
    * [ ] Run Full test suite (auto-detect all available tests and run them all.)
    * [x] Run individual test
    * [ ] Run all tests for a given component (I'd really like this!)

### CODE REVIEW

File: visreg.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+var wdio = require("webdriverio");
```
Comment (infiniteluke) [inline visreg.js]: I think we all use new enough versions of node that this could be a `const`

**Comment by infiniteluke** [comment]
I'd recommend a `test` dir with a `visual` subdir incase we ever add other types of tests.

**Comment by infiniteluke** [comment]
to make this compatible with yarn you should just have to `yarn upgrade` or `yarn install -D depName` for each dependency.

File: visreg.js (hunk 1 above)
Comment (ModulesUnraveled) [inline visreg.js]: Just this one? I'm super new to js, so I feel out of my depth. :) should all of the variables be constants?

**Comment by amazingrando** [comment]
@ModulesUnraveled The last work for this was in the spring. You okay with closing this PR and keeping the branch for future work?

**Comment by ModulesUnraveled** [comment]
Yeah, what we're doing on my current project is a lot better than this. So, I'll just create a new branch when it's time to set that up here

**Pull request closed without merging**